container.go

```diff
@@ -201,7 +201,15 @@ func (c Container) Call(function interface{}) error {
 		return err
 	}
 
-	reflect.ValueOf(function).Call(arguments)
+	result := reflect.ValueOf(function).Call(arguments)
+
+	// if the receiver returns an error then we'll forward
+	// the return value as the Call() result.
+	if len(result) == 1 && result[0].CanInterface() {
```

Comment:
This means library users can pass a receiver like this which I think is nice and intuitive with how the other methods in the library work:

```golang
c.Call(func(db Database) error {
    if err := db.Ping(); err != nil {
        return err
    }
    return nil
})
```